### PR TITLE
Service layer now has Interfaces

### PR DESCRIPTION
**Pull request by eddiejaoude**
| BC | No |
| ---- | ---- |
| Tests | Yes |
| Docs | No |

### CODE REVIEW

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java [hunk 1]
```diff
@@ -16,11 +16,11 @@
 public class AccountController {
 
     @Autowired
-    private UserService userService;
+    private UserServiceImpl userServiceImpl;
```
Comment (simoncasey) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java]: Should be coding against the Interface here and Autowired will pick the correct (and only) implementation:

```
@Autowired
private UserService userService;
```

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java [hunk 2]
```diff
@@ -34,14 +34,14 @@ public String profileSave(
             Model model
     ) {
 
-        this.userService.findByCurrentUser();
+        this.userServiceImpl.findByCurrentUser();
```
Comment (simoncasey) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java]: use of 'this'. There is no other userService in scope so this is not needed.

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/ProjectController.java [hunk 3]
```diff
@@ -19,11 +18,11 @@
 public class ProjectController {
 
     @Autowired
-    private ProjectService projectService;
+    private ProjectServiceImpl projectServiceImpl;
```
Comment (simoncasey) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/ProjectController.java]: as above for coding to interface with autowired

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/advice/CurrentUserAdvice.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 public class CurrentUserAdvice {
 
     @Autowired
-    UserService userService;
+    UserServiceImpl userServiceImpl;
```
Comment (simoncasey) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/advice/CurrentUserAdvice.java]: as above, also mark private

File: src/main/java/io/dashboardhub/pipelinedashboard/listener/AuthenticationListener.java [hunk 5]
```diff
@@ -13,18 +13,18 @@
 public class AuthenticationListener implements ApplicationListener<InteractiveAuthenticationSuccessEvent> {
 
     @Autowired
-    private UserService userService;
+    private UserServiceImpl userServiceImpl;
```
Comment (simoncasey) [inline src/main/java/io/dashboardhub/pipelinedashboard/listener/AuthenticationListener.java]: as above

File: src/main/java/io/dashboardhub/pipelinedashboard/service/ProjectServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProjectServiceImpl implements ProjectService {
+
+    @Autowired
+    private ProjectRepository projectRepository;
+
+    @Autowired
+    private UserServiceImpl userServiceImpl;
```
Comment (simoncasey) [inline src/main/java/io/dashboardhub/pipelinedashboard/service/ProjectServiceImpl.java]: code to interface

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java (hunk 1 above)
Comment (eddiejaoude) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java]: Yep, makes sense! I will update

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java (hunk 2 above)
Comment (eddiejaoude) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/AccountController.java]: I thought I removed it, but i will go through and check

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/ProjectController.java (hunk 3 above)
Comment (eddiejaoude) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/ProjectController.java]: I will update accordingly

File: src/main/java/io/dashboardhub/pipelinedashboard/controller/advice/CurrentUserAdvice.java (hunk 4 above)
Comment (eddiejaoude) [inline src/main/java/io/dashboardhub/pipelinedashboard/controller/advice/CurrentUserAdvice.java]: :+1: I will update accordingly

File: src/main/java/io/dashboardhub/pipelinedashboard/listener/AuthenticationListener.java (hunk 5 above)
Comment (eddiejaoude) [inline src/main/java/io/dashboardhub/pipelinedashboard/listener/AuthenticationListener.java]: I will update accordingly

File: src/main/java/io/dashboardhub/pipelinedashboard/service/ProjectServiceImpl.java (hunk 6 above)
Comment (eddiejaoude) [inline src/main/java/io/dashboardhub/pipelinedashboard/service/ProjectServiceImpl.java]: I will update accordingly

**Pull request merged**